### PR TITLE
[Debugger] Add arguments passing to application

### PR DESCRIPTION
**Pull request by xplicit**
For 'Run/Debug Application...' menu added ability to pass command-line
arguments, working directory and environment variables via selecting them
in dialog

### CODE REVIEW

**Comment by Therzok** [comment]
LGTM except for a few code styling issues. Can you also attach a screenshot of how this looks?

**Comment by xplicit** [comment]
md

**Comment by xplicit** [comment]
What exactly is wrong with codestyle? I'll fix it. I wonder why monodevelop does not understrike wrong codestyle with red wave line, it would be very helpful.

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs [hunk 1]
```diff
@@ -461,10 +461,16 @@ public static void SetNextStatement (string fileName, int line, int column)
 			NotifyLocationChanged ();
 		}
 
-		public static IProcessAsyncOperation Run (string file, IConsole console)
+		public static IProcessAsyncOperation Run (string file, string args, string workingDir, IDictionary<string,string> envVars, IConsole console)
```
Comment (slluis) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs]: Please don't break the public api. Add a new overload.

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugApplicationDialog.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+using System;
+using System.Collections.Generic;
+
+namespace MonoDevelop.Debugger
+{
+	public partial class DebugApplicationDialog : Gtk.Dialog
```
Comment (slluis) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugApplicationDialog.cs]: This class has to be internal.

**Comment by xplicit** [comment]
@slluis I've made proposed changes.

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugApplicationDialog.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+using System.Collections.Generic;
+
+namespace MonoDevelop.Debugger
+{
+	internal partial class DebugApplicationDialog : Gtk.Dialog
+	{
+		public DebugApplicationDialog ()
+		{
+			this.Build ();
+
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugApplicationDialog.cs]: Like this extra newline.

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugApplicationDialog.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+				return envVars;
+			}
+			set {
+				this.envvarlist1.LoadValues (value);
+			}
+		}
+
+		public string WorkingDirectory {
+			get {
+				return this.folderEntry.Path;
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugApplicationDialog.cs]: There should be on one line as:

```csharp
public string WorkingDirectory {
    get { ... }
    set { ... }
}

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugCommands.cs [hunk 5]
```diff
@@ -269,15 +270,42 @@ class DebugApplicationHandler: CommandHandler
...
-				if (IdeApp.ProjectOperations.CanDebugFile (dialog.SelectedFile))
-					IdeApp.ProjectOperations.DebugApplication (dialog.SelectedFile);
-				else
-					MessageService.ShowError (GettextCatalog.GetString ("The file '{0}' can't be debugged", dialog.SelectedFile));
+			var dlg = new DebugApplicationDialog ();
+
+			try {
+				bool isOK;
+
+				 while ((isOK = (MessageService.RunCustomDialog (dlg) == (int)Gtk.ResponseType.Ok))
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebugCommands.cs]: Nitpicky, but extra whitespace before while.

**Comment by xplicit** [comment]
@Therzok fixed

**Comment by xplicit** [comment]
Is there a chance to get this merged?

**Pull request merged**